### PR TITLE
Modify requisition URL alias (GLT-3830)

### PR DESCRIPTION
**Pull request by hyeonjeongkwon**
Jira ticket or GitHub issue: GLT-3830

- [add_alias_requisition_URL ] Includes a change file

### CODE REVIEW

File: changes/add_alias_requisition_URL_ [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Added Alias to requisiton URL
+- ex) https://miso.oicr.on.ca/miso/requisition/alias/REQ-1 where REQ-1 is the alias
```
Comment (djcooke) [inline changes/add_alias_requisition_URL_:2]: * This sounds like the existing URL was changed, but you've actually added a new one. Please reword - something like "Added a new URL pattern for linking to requisitions by alias"
* In the example, exclude the OICR-specific part (start at `/miso/`)

File: miso-web/pom.xml [hunk 2]
```diff
@@ -1,918 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (djcooke) [inline miso-web/pom.xml:1]: This file should not be deleted

**Review by djcooke — COMMENTED** [review]
It looks like there are no changes in `SampleRestController` and `LibraryRestController` except formatting?

File: changes/add_alias_requisition_URL_ [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Added a new URL pattern for linking to requisitions by alias
+- ex) /miso.oicr.on.ca/miso/requisition/alias/REQ-1 where REQ-1 is the alias
```
[the remark this answers is not in the document]
Comment (hyeonjeongkwon) [reply changes/add_alias_requisition_URL_:2]: Will modify right now

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/RequisitionController.java [hunk 4]
```diff
@@ -87,17 +87,31 @@ public ModelAndView edit(@PathVariable long id, ModelMap model) throws IOExcepti
...
+    return setupEditForm(requisition, model);
+  }
 
-    List<Sample> samples = sampleService.list(0, 0, false, "id", PaginationFilter.requisitionId(id));
+  @GetMapping("/alias/{alias}")
+  public ModelAndView edit(@PathVariable String alias, ModelMap model) throws IOException {
+    Requisition requisition = requisitionService.getByAlias(alias);
+    if (requisition == null) {
+      throw new NotFoundException("No requisition found for ID: " + alias);
+    }
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/RequisitionController.java:98]: Can move this check into `setupEditForm` as well

**Review by djcooke — APPROVED** [review]

**Review by callunity — APPROVED** [review]

**Pull request merged**